include/deal.II/base/symmetric_tensor.h

```diff
@@ -1215,17 +1215,15 @@ namespace internal
       {
       case 1:
         return data[0] * sdata[0];
-      case 2:
-        return (data[0] * sdata[0] +
-                data[1] * sdata[1] +
-                NumberType<Number>::value(2.0) * data[2] * sdata[2]);
       default:
         // Start with the non-diagonal part to avoid some multiplications by
         // 2.
         Number sum = data[dim] * sdata[dim];
         for (unsigned int d=dim+1; d<(dim*(dim+1)/2); ++d)
           sum += data[d] * sdata[d];
-        sum *= NumberType<Number>::value(2.0);
+        sum += sum; // sum = sum * 2.;
```

Comment:
Alright, no problem. I think it's slightly harder to read, but it doesn't matter at all and the comment explains it.

And of course you're right, multiply-by-two is an addition on the exponent, not a bit shift :-)